core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1460,39 +1460,41 @@ public boolean removePartition(long timestamp) {
-            final int prevIndex = index - 1;
-            final long prevTimestamp = txWriter.getPartitionTimestamp(prevIndex);
-            final long newTransientRowCount = txWriter.getPartitionSize(prevIndex);
-            try {
-                setPathForPartition(path.trimTo(rootLen), partitionBy, prevTimestamp, false);
-                TableUtils.txnPartitionConditionally(path, txWriter.getPartitionNameTxn(prevIndex));
-                readPartitionMinMax(ff, prevTimestamp, path, metadata.getColumnName(metadata.getTimestampIndex()), newTransientRowCount);
-            }
-            finally {
-                path.trimTo(rootLen);
-            }
+                // calculate new transient row count and max timestamp
+                final int prevIndex = index - 1;
+                final long prevTimestamp = txWriter.getPartitionTimestamp(prevIndex);
+                final long newTransientRowCount = txWriter.getPartitionSize(prevIndex);
+                try {
+                    setPathForPartition(path.trimTo(rootLen), partitionBy, prevTimestamp, false);
+                    TableUtils.txnPartitionConditionally(path, txWriter.getPartitionNameTxn(prevIndex));
+                    readPartitionMinMax(ff, prevTimestamp, path, metadata.getColumnName(metadata.getTimestampIndex()), newTransientRowCount);
+                } finally {
+                    path.trimTo(rootLen);
+                }
 
-            columnVersionWriter.removePartition(timestamp);
-            txWriter.beginPartitionSizeUpdate();
-            txWriter.removeAttachedPartitions(timestamp);
-            // max is updated upon finishing the transaction, the value was loaded by readPartitionMinMax
-            txWriter.finishPartitionSizeUpdate(txWriter.getMinTimestamp(), attachMaxTimestamp);
-            txWriter.bumpTruncateVersion();
+                columnVersionWriter.removePartition(timestamp);
+                txWriter.beginPartitionSizeUpdate();
+                txWriter.removeAttachedPartitions(timestamp);
+                txWriter.finishPartitionSizeUpdate(txWriter.getMinTimestamp(), attachMaxTimestamp);
+                txWriter.bumpTruncateVersion();
 
-            columnVersionWriter.commit();
-            txWriter.setColumnVersion(columnVersionWriter.getVersion());
-            txWriter.commit(defaultCommitMode, denseSymbolMapWriters);
+                columnVersionWriter.commit();
+                txWriter.setColumnVersion(columnVersionWriter.getVersion());
+                txWriter.commit(defaultCommitMode, denseSymbolMapWriters);
 
-            closeActivePartition(true);
-            row = regularRow;
-            openPartition(prevTimestamp);
-            setAppendPosition(newTransientRowCount, false);
+                closeActivePartition(true);
+                row = regularRow;
+                openPartition(prevTimestamp);
+                setAppendPosition(newTransientRowCount, false);
```

Comment:
Should we call `safeDeletePartitionDir(timestamp, partitionNameTxn);` here?

Reply:
extracted so that it is done